src/Centreon/Application/Controller/PlatformTopologyController.php

```diff
@@ -0,0 +1,188 @@
+ *
+ * @package Centreon\Application\Controller
+ */
+class PlatformTopologyController extends AbstractController
+{
+    /**
+     * @var array<string> Allowed server types
+     */
+    private $allowedType = ["Central", "Poller", "Remote", "MAP", "MBI"];
+
+    /**
+     * @var PlatformTopologyServiceInterface
+     */
+    private $platformTopologyService;
+
+    /**
+     * PlatformTopologyController constructor
+     * @param PlatformTopologyServiceInterface $platformTopologyService
+     */
+    public function __construct(PlatformTopologyServiceInterface $platformTopologyService)
+    {
+        $this->platformTopologyService = $platformTopologyService;
+    }
+
+    /**
+     * Validate platform topology data according to json schema
+     *
+     * @param array<mixed> $platformToAdd data sent in json
+     * @param string $schemaPath
+     * @return void
+     * @throws PlatformTopologyException
+     */
+    private function validatePlatformTopologySchema(array $platformToAdd, string $schemaPath): void
+    {
+        $platformTopologySchemaToValidate = Validator::arrayToObjectRecursive($platformToAdd);
+        $validator = new Validator();
+        $validator->validate(
+            $platformTopologySchemaToValidate,
+            (object) ['ref' => 'file://' . $schemaPath],
+            Constraint::CHECK_MODE_VALIDATE_SCHEMA
+        );
+
+        if (!$validator->isValid()) {
+            $message = '';
+            foreach ($validator->getErrors() as $error) {
+                $message .= sprintf("[%s] %s\n", $error['property'], $error['message']);
+            }
+            throw new PlatformTopologyException($message);
+        }
+    }
+
+    /**
+     * Entry point to register a new server
+     *
+     * @param Request $request
+     * @return View
+     * @throws PlatformTopologyException
+     */
+    public function addPlatformToTopology(Request $request): View
+    {
+        // check user rights
+        $this->denyAccessUnlessGrantedForApiConfiguration();
+
+        // get http request content
+        $platformToAdd = json_decode((string) $request->getContent(), true);
+        if (!is_array($platformToAdd)) {
+            return $this->view([
+                'code' => Response::HTTP_BAD_REQUEST,
+                'message' => sprintf(
+                    _('Error when decoding sent data')
+                )
+            ]);
+        }
+
+        // check data consistency
+        $this->validatePlatformTopologySchema(
+            $platformToAdd,
+            $this->getParameter('centreon_path')
+            . 'config/json_validator/latest/Centreon/PlatformTopology/AddServer.json'
+        );
+
+        // sanitize data
+        $platformToAdd['ip_address'] = filter_var($platformToAdd['ip_address'], FILTER_VALIDATE_IP);
+        $platformToAdd['server_name'] = filter_var($platformToAdd['server_name'], FILTER_SANITIZE_STRING);
```

Comment:
not useful since string is already checked in json schema

Reply:
I added it to sanitize the string sent in the data
As seen together, I resolve this one